task_sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -286,13 +286,13 @@ def exit(n: int) -> NoReturn:
 
 @attrs.define()
 class WatchedSubprocess:
-    id: UUID
+    id: UUID | None
     pid: int
 
     stdin: BinaryIO
     """The handle connected to stdin of the child process"""
 
-    client: Client
+    client: Client | None
```

Comment:
What broke with leaving this as it was?

I wonder if instead of doing this change it might be better to have three classes -- the "base" WatchedSubproces, and then create a new class like:

```python
class ActivitySubprocess(WatchedSubprocess):
    id: UUID
    client: Client

    def _on_child_started(self, ti: TaskInstance, path: str | os.PathLike[str], requests_fd: int):
        ...
```

Reply:
no client is provided by dag processor it was explicitly given _none_; so clearly the base class is ok with that. it's only for task execution scenario that it is used and depended on. that's why the type checking asserts

Reply:
Yeah but what broke by leaving it as it was?